src/main/java/com/zup/ecommerce/controllers/ProductController.java

```diff
@@ -0,0 +1,46 @@
+package com.zup.ecommerce.controllers;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import com.zup.ecommerce.models.Product;
+import com.zup.ecommerce.services.ProductService;
+
+import java.util.List;
+
+
+@RestController
+@RequestMapping("api/product")
+public class ProductController {
+    
+    private final ProductService productService;
+
+    public ProductController(ProductService productService){
+        this.productService = productService;
+    }
+
+    @PostMapping
+    public ResponseEntity<Product> createProduct(@RequestBody Product product) {
+        Product createdProduct = productService.createProduct(product);
+        return new ResponseEntity<>(createdProduct, HttpStatus.CREATED);
+    }
```

Comment:
Detalhe besta, mas a anotação @RestController torna o `ResponseEntity` opcional, pois ela já diz para o SpringBoot que os retornos dessa classe são para uma API REST. Dá para reescrever assim:

```suggestion
    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public Product createProduct(@RequestBody Product product) {
        return productService.createProduct(product);
    }
```

Fica um pouco mais enxuto.

Originalmente, o SpringBoot tinha apenas a anotação `@Controller` , essa sim precisa explicitamente de um `ResponseEntity`. A `@RestController` foi introduzida recentemente justamente pra facilitar o código.